appeals/web/src/server/appeals/appeal-details/interested-party-comments/add-ip-comment/__tests__/add-ip-comment.test.js

```diff
@@ -0,0 +1,276 @@
+import { parseHtml } from '@pins/platform';
+import nock from 'nock';
+import supertest from 'supertest';
+import { createTestEnvironment } from '#testing/index.js';
+import { appealData } from '#testing/app/fixtures/referencedata.js';
+
+const { app, installMockApi, teardown } = createTestEnvironment();
+const request = supertest(app);
+const baseUrl = '/appeals-service/appeal-details';
+
+describe('add-ip-comment', () => {
+	beforeEach(installMockApi);
+	afterEach(teardown);
+
+	describe('GET /add', () => {
+		it('should redirect to /add/ip-details', () => {
+			return new Promise((resolve) => {
+				request
+					.get(`${baseUrl}/2/interested-party-comments/add`)
+					.expect(302)
+					.expect('Location', `./add/ip-details`)
+					.end(resolve);
+			});
+		});
+	});
+
+	describe('GET /add/ip-details', () => {
+		const appealId = 2;
+
+		/** @type {*} */
+		let pageHtml;
+
+		beforeAll(async () => {
+			nock('http://test/')
+				.get(`/appeals/${appealId}`)
+				.reply(200, { ...appealData, appealId, appealStatus: 'awaiting_transfer' });
```

Comment:
I assume this is to ensure we can test the page before reaching the correct status. `awaiting_transfer` is a bit of a weird one, I would probably have chosen `lpa_questionnaire` which is when we start getting the comments notifications. In any case, no big deal, we can always change as we progress.

Reply:
Good spot, I actually copied that snippet from another file where I'm guessing that value was more relevant. I've removed it so it just defaults to `lap_questionnaire`. 👍